### PR TITLE
Build VPA releases in docker

### PR DESCRIPTION
**Pull request by bskiba**
@MaciekPytel

### CODE REVIEW

File: vertical-pod-autoscaler/pkg/recommender/Makefile [hunk 1]
```diff
@@ -1,33 +1,59 @@
...
-	docker build --pull -t ${REGISTRY}/vpa-recommender:${TAG} .
-	docker push ${REGISTRY}/vpa-recommender:${TAG}
+ifndef TAG
+	ERR = $(error TAG is undefined)
+	$(ERR)
+endif
+	docker push ${REGISTRY}/${FULL_COMPONENT}:${TAG}
+
+docker-builder:
+	docker build -t autoscaling-builder ../../../builder
```
Comment (aleksandra-malinowska) [inline vertical-pod-autoscaler/pkg/recommender/Makefile]: Using the same builder for CA and VPA will mean we always have to do things like upgrade Go version simultaneously etc. Do we want to do this?

**Comment by bskiba** [comment]
@aleksandra-malinowska @MaciekPytel does this look resonable now?

File: vertical-pod-autoscaler/pkg/admission-controller/Makefile [hunk 2]
```diff
@@ -1,33 +1,59 @@
...
+	ERR = $(error TAG is undefined)
+	$(ERR)
+endif
+	docker push ${REGISTRY}/${FULL_COMPONENT}:${TAG}
+
+docker-builder:
+	docker build -t vpa-autoscaling-builder ../../builder
+
+build-in-docker: clean docker-builder
+	docker run -v `pwd`/../..:/gopath/src/k8s.io/autoscaler/vertical-pod-autoscaler vpa-autoscaling-builder:latest bash -c 'cd /gopath/src/k8s.io/autoscaler/vertical-pod-autoscaler && make build-binary -C pkg/admission-controller'
```
Comment (aleksandra-malinowska) [inline vertical-pod-autoscaler/pkg/admission-controller/Makefile]: Is it intentional to not parametrize `admission-controller` here?

Comment (aleksandra-malinowska) [reply vertical-pod-autoscaler/pkg/admission-controller/Makefile]: Looking at the rest of the Makefiles, I guess my question is more if there's any difference between them except this? And why parametrize them using COMPONENT if they're forked anyway?

Comment (bskiba) [reply vertical-pod-autoscaler/pkg/admission-controller/Makefile]: No specific reason I guess I can inline everywhere if that's more readable

Comment (aleksandra-malinowska) [reply vertical-pod-autoscaler/pkg/admission-controller/Makefile]: Ack, as you prefer

**Pull request merged**